src/mbgl/renderer/render_layer.hpp

```diff
@@ -58,6 +66,8 @@ class RenderLayer {
     // Checks whether this layer can be rendered.
     bool needsRendering(float zoom) const;
 
+    virtual void uploadBuckets(gl::Context&, RenderSource* source);
+    virtual void render(Painter& , PaintParameters& , RenderSource* source);
```

Comment:
The `RenderSource*` parameters of both of these isn't used; we should remove it.